### PR TITLE
Fix dnslayer memory overflow when add Resource

### PR DESCRIPTION
**Pull request by aengusjiang**
relative issue: https://github.com/seladb/PcapPlusPlus/issues/1137

### CODE REVIEW

**Comment by egecetin** [comment]
@aengusjiang I have a general question about the buffer size. You recommended buffer size 2048 or 4096 bytes. I checked the RFC4034 but I can't find any information about the maximum size. It only mentions, it depends on the algorithm (SHA1 etc.) but do you know is there any theoretical/practical limit for the DNSKEY field?

**Comment by egecetin** [comment]
I understand now, thanks.

**Comment by egecetin** [comment]
Looks like the tests are failing after change.
image

**Comment by aengusjiang** [comment]
I understand. I remove the invalid check dataLength. In fact, it just pass 0.

File: Packet++/src/DnsLayer.cpp [hunk 1]
```diff
@@ -561,7 +561,7 @@ DnsResource* DnsLayer::addResource(DnsResourceType resType, const std::string& n
 DnsQuery* DnsLayer::addQuery(const std::string& name, DnsType dnsType, DnsClass dnsClass)
 {
 	// create new query on temporary buffer
-	uint8_t newQueryRawData[256];
+	uint8_t newQueryRawData[4096];
```
Comment (seladb) [inline Packet++/src/DnsLayer.cpp:564]: isn't DNSKey a resource record and not a query?
Are there queries that exceed 256 bytes?

File: Packet++/src/DnsLayer.cpp
Comment (seladb) [inline Packet++/src/DnsLayer.cpp:1]: Is it possible to add a test that adds DNSKey with a length of more than 256 bytes?

File: Packet++/src/DnsLayer.cpp (hunk 1 above)
Comment (aengusjiang) [reply Packet++/src/DnsLayer.cpp:564]: DNSKey  is an answer record, here maybe needn't change to 4096. I will try revert it.

File: Packet++/src/DnsLayer.cpp
Comment (aengusjiang) [reply Packet++/src/DnsLayer.cpp:1]: Good idea. I will add later.

**Comment by seladb** [comment]
@aengusjiang please let me know if you are planning to address the comments on this PR so we can merge it

File: Packet++/src/DnsResourceData.cpp [hunk 2]
```diff
@@ -218,7 +218,7 @@ bool GenericDnsResourceData::toByteArr(uint8_t* arr, size_t& arrLength, IDnsReso
 {
 	if (m_DataLen == 0 || m_Data == nullptr)
 	{
-		PCPP_LOG_ERROR("Input data is null or illegal");
+		PCPP_LOG_ERROR("Input data is null or illegal" << "|m_DataLen:" << m_DataLen);
```
Comment (seladb) [inline Packet++/src/DnsResourceData.cpp:221]: This change is probably not needed, but we can fix it later

**Pull request merged**